### PR TITLE
fix: precise fee estimation and oscillation protection (Supersedes #195)

### LINKED ISSUE #63: Wallet.sendOnchain should compute the tx size in order to estimate fees
The tx size now is hardcoded, it should use `TxWeightEstimator` class.

https://github.com/arkade-os/ts-sdk/blob/master/src/wallet/onchain.ts#L113

### PR DESCRIPTION
**Pull request by jgmcalpine**
### Description
This PR implements precise fee estimation using address decoding and resolves the "dust oscillation" loop reported by CodeRabbit to close #63.

**Reason for new PR:**
This supersedes PR #195 due to inactivity. I have built upon the original work by @shubertm and added fixes requested by @louisinger and CodeRabbit which were blocking the merge.

### Changes
1. **Precise Script Sizes:** Replaced string heuristics with `@scure/btc-signer` address decoding. This ensures correct fee calculation for Legacy, Segwit, and Taproot inputs/outputs (Fixes @louisinger feedback).
2. **Oscillation Fix:** Refactored `estimateFeesAndSelectCoins` to handle the "dust paradox" where removing a change output drops the fee below the budget. This prevents infinite loops/timeouts.
3. **Tests:** Added unit tests ensuring:
   - Segwit outputs are calculated as smaller than Taproot.
   - The fee estimator converges successfully in dust-boundary edge cases.

Co-authored-by: @shubertm 

## Summary by CodeRabbit

* **New Features**
  * Enhanced transaction-size estimator with more accurate address-based output sizing and varint sizing helper.

* **Behavior Changes**
  * Send, bump and unroll flows use convergence-based fee/coin selection; wallet avoids creating change outputs below the dust threshold.
  * Dust threshold exposed for consistent handling.

* **Bug Fixes / Refinements**
  * Minor on-chain validation/refactor to ensure explicit boolean returns in block-tip checks.

* **Tests**
  * Expanded wallet tests for insufficient funds, dust validation, change-edge cases, and fee/size scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/providers/onchain.ts [hunk 1]
```diff
@@ -350,14 +350,16 @@ function isValidBlocksTip(
     return (
         Array.isArray(tip) &&
         tip.every((t) => {
-            t &&
+            return (
```
Comment (pietro909) [inline src/providers/onchain.ts]: This is quite an important fix: `isValidBlocksTip` has been returning `false` all the time.

File: src/utils/txSizeEstimator.ts [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
 };
 
+/**
+ * Calculates the byte size required to store a variable-length integer (VarInt).
+ * Bitcoin uses VarInts to compact integer data (like array lengths).
+ *
+ * @param n - The integer value to check
+ * @returns The size in bytes (1, 3, 5, or 9)
+ */
+const getVarIntSize = (n: number): number => {
```
Comment (shubertm) [inline src/utils/txSizeEstimator.ts]: :+1:
